File: Java/016.java
import java.math.BigInteger;

public class Problem016 {

    public static int sumDigits(BigInteger num) {
        String s = num.toString();
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            res += Character.getNumericValue(s.charAt(i));
        }
        return res;
    }

    public static void main(String[] args) {
        BigInteger num = new BigInteger("2").pow(1000);
        System.out.println(sumDigits(num));
    }
}

File: Rust/016.rs
use num_bigint::BigUint;

fn sum_digits(num: &BigUint) -> u32 {
    num.to_string().chars().map(|c| c.to_digit(10).unwrap()).sum()
}

fn main() {
    let base = BigUint::from(2u32);
    let power = 1000;
    let num = base.pow(power);
    println!("{}", sum_digits(&num));
}
